### ISSUE TITLE
Final block family tweaks needed before v2.0.0 release

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/Terasology#3374]
Follow-up to #3343 to cover just the remaining problems to resolve before we can release v2.0.0, making this issue a Blocker, yay!

- [x] A chest block places normally, facing the player who placed it. However a stair block (a stack exists inside the starting goodie chest) *always* faces in the same direction regardless of how placed. Maybe there's something wrong with the direction handling for that family?
- [ ] Review comments still to address in https://github.com/Terasology/MarcinScIncubator/pull/3#pullrequestreview-118538219 about use of `super`
- [ ] Better javadoc including review comments at https://github.com/MovingBlocks/Terasology/pull/3343#pullrequestreview-116639819
- [ ] ~Some StructureTemplates in GooeysQuest may still be acting funny. I had a Dwarf Hall placed where the staircase down seemed intact (other than the directionally-challenged stairs) but the spacious halls were full of stone bars going across from side to side. And a dungeon city had 3 normal looking buildings and one that seemingly started half a dozen blocks up the remaining wall. The fix in https://github.com/Terasology/StructureTemplates/commit/0c6e6a9f96cdffd4a8cdde165646d9be04d0811d might not be good enough~ - moved to #3390
- [x] Rails needs block family updates
- [x] Initial Rails fix left sloped rails still broken - need to fix further sometime
- [ ] Minesweeper needs block family updates
- [x] ItemPipes needs block family updates
- [ ] TutorialAssetSystem (and probably its wiki as well) needs the updates. 
- [ ] https://github.com/Terasology/BlockFamilyTutorial is also a thing. Does it work? Could it be updated to latest? Should it be merged into the TutorialAssetSystem?
- [ ] TTA (it isn't a template anymore, enable its module + pick its world manually) *does* show its trees from AnotherWorldPlants OK now, with the "pipes" as branches inside the canopy of trees. However, the `.tree` files in AWP still have the very verbose connection info in them, along with a block family reference. When I tried to delete all that the branches went to full trunk mode again. Huh?
- [x] Terasology/Machines#25 - also clashes with the changes in #3221 and is a bit more work than I can put in right now (complex family with all faces and an archetype, logic defines it all over several passes, hard to readily tweak)
- [x] Terasology/Signalling#17 - PR not yet merged, was prepared before the block family code was final
- [x] ModularComputers has a small compile error, but can't be tested without Signalling fixed
- [ ] Sample needs updates and has a bunch of block family related PRs
- [x] JoshariasSurvival has pipes that do not work, although they were broken even *before* the overhaul, so ..
- [x] https://github.com/Terasology/AdditionalRails/pull/27
- [ ] #2907 needs some review in light of this PR being merged

The more we can get fixed the better! Will roll what we can into an Alpha 10 / v2 Release Candidate for the play test in a week.

Maybe we can get some more help from @pollend  @PS-Soundwave @Steampunkery @anuar2k @NicholasBatesNZ ... who else? :-)

Oh! And all that didn't cover what was actually fixed, but could use more testing! The following modules were affected (in addition to the above):

* HjerlHede
* FunnyBlocks
* GooeysQuests
* CopperAndBronze (TTA - unmaintained)
* AdventureAssets (Lost)
* AdditionalRails
* EdibleFlora (JS)
* JoshariasSurvival
* Lost
* ManualLabor (JS)
* StoneCrafting (NeoTTA)
* StructureTemplates (GooeysQuests)
* Valentines
* WoodAndStone (TTA)
* WoodCrafting (NeoTTA)
* https://github.com/Terasology/Fences/pull/8 (StaticCities)
* https://github.com/Terasology/StructuralResources/pull/3
* https://github.com/Terasology/MultiBlock/pull/6 (TTA, NeoTTA)
* https://github.com/Terasology/IRLCorp/pull/16 (JS)
* https://github.com/Terasology/ManualLabor/pull/23 (JS)
* https://github.com/Terasology/GrowingFlora/pull/12 (TTA
* LightAndShadow
* LightAndShadowResources (L&S)
* https://github.com/Terasology/MarcinScIncubator/pull/3 (TTA)
* AnotherWorldPlants (TTA/GF - fixed "pipe" branches)

**Comment by pollend** [comment]
For Rails I need to check diagnol block but the : https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/world/block/entity/neighbourUpdate/NeighbourBlockFamilyUpdateSystem.java

only checks in the 4 cardinal directions. What could be done to extend this. I was thinking of either adding more to: https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/world/block/family/UpdatesWithNeighboursFamily.java or just having the NeighbourBlockFamilyUpdateSystem check all directions in the 3x3 area around a given block.

**Comment by PS-Soundwave** [comment]
Changing current functionality could both cause unforeseen bugs and performance issues, I think its most appropriate to create a custom extended system.

**Comment by pollend** [comment]
I'm not sure what you mean specifically. I see a couple options I could go from here. I could add an additional getter function that denotes which blocks to update or just remove the implementation from multi-connect. The implementation could also be denoted in block.java. Something that also occurred to me is implement this as a component.

The main problem is that block updates don't at the moment work diagonally. I can't really see a problem with having them work that way. Might just be better update all the blocks in a 3x3 area around a given block in NeighbourBlockFamilyUpdateSystem.

**Comment by Cervator** [comment]
Removing Blocker from this issue as a pile of fixes have been done. special thanks to @pollend and @Josharias for most of them! :-)

The short-term fix for Rails is in place, then we can start another iteration on block family overhaul after Alpha 10 is merged to master shortly

Closing this as _mostly_ done, splitting out / moving the remaining issues
